eclipse/che-theia
# chore(devfile): che-theia devfile to work on plugins

Signed-off-by: Sun Tan <sutan@redhat.com>

### What does this PR do?
A devfile for dogfooding. At the moment it covers only the plugins dev flow.

### Screenshot/screencast of this PR
![Selection_208](https://user-images.githubusercontent.com/650571/107392774-4eb84700-6afa-11eb-8dcc-2558f68ad9a0.png)



### What issues does this PR fix or reference?
Part of https://github.com/eclipse/che/issues/18763

### How to test this PR?
- create a new workspace out of this url: https://github.com/eclipse/che-theia/tree/devfile-plugin
- It should clone `che-theia` repo
- choose any plugin to work on: for instance the welcome-plugin
- open any file located in `/projects/che-theia/plugins/welcome-plugin` for instance `src/welcome-plugin.ts`
- run the command `build ... current yarn project`: it should build the project (may take a while the first time)
- run the command `test-watch ... current yarn project`: it should fail as there is no tests in this plugin. For plugins with tests, it should run the test in watch mode: any changes on files would restart the test and warn the developer if a test is failling.
- do a change in `src/welcome-plugin.ts`, for instance:
  ![Selection_203](https://user-images.githubusercontent.com/650571/107383027-d8fbad80-6af0-11eb-97e3-c162480fb5cc.png)
- run the command `build current plugin ... and run with che-theia` it should build the plugin (without format/lint/test checks) and start che-theia. it should only take few seconds
- see your changes in the welcome page
  ![Selection_204](https://user-images.githubusercontent.com/650571/107383714-8f5f9280-6af1-11eb-89f4-c563e9c3d465.png)
- Ctrl-C to stop the server
- do a change in `src/welcome-plugin.ts`, for instance:
  ![Selection_205](https://user-images.githubusercontent.com/650571/107383888-bcac4080-6af1-11eb-98f1-afab5a418a4d.png)
- rerun the command `build current plugin ... and run with che-theia`
- see your changes in the welcome page
  ![Selection_206](https://user-images.githubusercontent.com/650571/107384448-4825d180-6af2-11eb-90c4-d044d8633a71.png)


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
